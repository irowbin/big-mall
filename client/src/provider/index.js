export * from './cart/cart-provider'
export * from './root/root-provider'
export * from './user/user-provider'
