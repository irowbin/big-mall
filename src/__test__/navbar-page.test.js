import {shallow, mount} from 'enzyme'
import {render, fireEvent, cleanup} from '@testing-library/react'
import Navbar from '../pages/navbar/navbar.component'
import {UserContext} from '../provider/index'

describe('Navbar page', () => {
  afterEach(cleanup)
  let wrapper
  const userContextValues =
    {
      state:  {currentUser:{displayName: 'test'}},
      dispatch: jest.fn()
    }

  beforeEach(()=>{
    const NavComponent = () =>(
      <UserContext.Provider value={{ ...userContextValues}} >
        <Navbar />
      </UserContext.Provider>
    )
    wrapper = shallow(<NavComponent />)
  })

  it('should match snapshot', ()=> {
    expect(wrapper).toMatchSnapshot()
  })

  it('should define initial state to the context hook',() =>{
    const navWrapper = wrapper.find(Navbar)
    const currentUser = navWrapper.find('span')
    console.log(currentUser)
   // expect(currentUser.text()).toBe('test')
  })
})
