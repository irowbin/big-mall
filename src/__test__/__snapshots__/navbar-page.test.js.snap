// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar page should match snapshot 1`] = `ShallowWrapper {}`;
